src/game.h

```diff
@@ -9,6 +9,7 @@
 #include "int_id.h"
 #include "cursesdef.h"
 #include "pimpl.h"
+#include "monstergenerator.h"
```

Comment:
You don't need to drag in this header if all you need is the declaration (not even the definition) of `species_type`, just add a forward declaration of it.

Reply:
Can you help me with this one? I removed the include, added `struct species_type;`, but compiler complains about  `invalid use of incomplete type 'using species_id = class string_id<species_type> {aka class string_id<species_type>}'`.

Reply:
> Can you help me with this one? I removed the include, added struct species_type;, but compiler complains about invalid use of incomplete type 'using species_id = class string_id<species_type> {aka class string_id<species_type>}'.

You need to include "string_id.h" here. All other usage of `string_id` in this header (e.g. as reference parameter) does not require it to be a complete type, but you have added an instance of it (as default value for the parameter) and that requires it to be a complete type.

Note that the include of "monstergenerator.h" is still unnecessary.